gateway/src/main.rs

```diff
@@ -35,7 +36,42 @@ fn print_key(key: &secrecy::SecretString) {
     println!("{}", key.expose_secret());
 }
 
-async fn handle_create_api_key() -> Result<(), Box<dyn std::error::Error>> {
+/// Specify the expiration policy for newly-created API keys.
+#[derive(Debug, Clone)]
+enum ApiKeyExpirySetting {
+    /// Denotes an API key that has no expiration datetime.
+    Infinite,
+    /// Denotes an API key that is set to expire at the corresponding datetime.
+    Finite(DateTime<Utc>),
+}
+
+impl From<&str> for ApiKeyExpirySetting {
+    fn from(v: &str) -> Self {
+        if v == "infinite" {
+            Self::Infinite
+        } else {
+            v.parse::<DateTime<Utc>>()
+                .map_or(Self::Infinite, Self::Finite)
```

Comment:
We should error if it fails to parse rather than silently falling back to infinite. Again not sure how `TryFrom` interacts with `clap` so we might need to parse manually.

Reply:
Agree. Will see what we can do.